lib/matplotlib/tests/test_datetime.py

```diff
@@ -318,11 +318,16 @@ def test_spy(self):
         fig, ax = plt.subplots()
         ax.spy(...)
 
-    @pytest.mark.xfail(reason="Test for stackplot not written yet")
     @mpl.style.context("default")
     def test_stackplot(self):
-        fig, ax = plt.subplots()
-        ax.stackplot(...)
+        mpl.rcParams["date.converter"] = 'concise'
+        N = 10
+        fig, ax = plt.subplots(constrained_layout=True)
```

Comment:
This should use the new argument:
```suggestion
        fig, ax = plt.subplots(layout='constrained')
```
I'd also move it down near `ax.stackplot` to not split it by the data creation code.